pkg/controller/issuer/certificate/reconciler.go

```diff
@@ -644,6 +658,26 @@ func (r *certReconciler) checkDomainRangeRestriction(issuerDomains *api.DNSSelec
 	return nil
 }
 
+func (r *certReconciler) getDuration(cert *api.Certificate) (*time.Duration, error) {
+	if cert.Spec.Duration == nil {
+		return nil, nil
+	}
+	duration := cert.Spec.Duration.Duration
+	if duration < 2*r.renewalWindow {
+		return nil, fmt.Errorf("certificate duration must be greater than %v", 2*r.renewalWindow)
+	}
+	return ptr.To(duration), nil
+}
+
+func (r *certReconciler) validateCertDuration(duration *time.Duration, caKeyPair *legobridge.TLSKeyPair) error {
+	caNotAfter := caKeyPair.Cert.NotAfter
```

Comment:
Ensure that duration is not nil.
```suggestion
        if duration == nil {
          return nil
        }
	caNotAfter := caKeyPair.Cert.NotAfter
```

Reply:
Fixed with: 2ba05b9c4b894b6993f77ca8b2ace578e86a0868